examples/speaker_diarization/speaker_diarization/evaluator.py

```diff
@@ -0,0 +1,224 @@
+        IdentificationRecall=IdentificationRecall()(reference, inference),
+
+        WordErrorRate=wer(gt_text, inf_text),
+        CharacterErrorRate=cer(gt_text, inf_text),
+
+        IdentificationError=generate_fp(gt, inf, identification=True),
+        MissedSpeechError=generate_tp(gt, inv(inf, gt), identification=False),
+    )
+
+
+def compute_aggregate_metrics(
+    test_samples_metrics: List[TestSampleMetric],
+    test_samples: List[TestSample],
+    ground_truths: List[GroundTruth],
+    inferences: List[Inference],
+) -> TestCaseMetric:
+    n_samples = len(test_samples)
+    return TestCaseMetric(
+        DiarizationErrorRate=sum([metric.DiarizationErrorRate for metric in test_samples_metrics]) / n_samples,
+        JaccardErrorRate=sum([metric.JaccardErrorRate for metric in test_samples_metrics]) / n_samples,
+        DiarizationPurity=sum([metric.DiarizationPurity for metric in test_samples_metrics]) / n_samples,
+        DiarizationCoverage=sum([metric.DiarizationCoverage for metric in test_samples_metrics]) / n_samples,
+
+        DetectionAccuracy=sum([metric.DetectionAccuracy for metric in test_samples_metrics]) / n_samples,
+        DetectionPrecision=sum([metric.DetectionPrecision for metric in test_samples_metrics]) / n_samples,
+        DetectionRecall=sum([metric.DetectionRecall for metric in test_samples_metrics]) / n_samples,
+
+        IdentificationErrorRate=sum([metric.IdentificationErrorRate for metric in test_samples_metrics]) / n_samples,
+        IdentificationPrecision=sum([metric.IdentificationPrecision for metric in test_samples_metrics]) / n_samples,
+        IdentificationRecall=sum([metric.IdentificationRecall for metric in test_samples_metrics]) / n_samples,
+    )
+
+
+def compute_score_distribution_plot(
+    score: str,
+    metrics: List[Union[TestSampleMetric, Inference]],
+    binning_info: Optional[Tuple[float, float, float]] = None,  # start, end, num
+    logarithmic_y: bool = False,
+    logarithmic_x: bool = False,
+) -> Histogram:
+    scores = [getattr(m, score) for m in metrics]
+    if logarithmic_x:
+        bins = np.logspace(*binning_info, base=2)
+    else:
+        bins = np.linspace(*binning_info)
+
+    hist, _ = np.histogram(scores, bins=bins)
+    return Histogram(
+        title=f"Distribution of {score}",
+        x_label=f"{score}",
+        y_label="Count",
+        buckets=list(bins),
+        frequency=list(hist),
+        y_config=AxisConfig(type="log") if logarithmic_y else None,
+        x_config=AxisConfig(type="log") if logarithmic_x else None,
+    )
+
+
+def compute_metric_vs_metric_plot(
+    x_metric: str,
+    y_metric: str,
+    x_metrics: List[Union[TestSampleMetric, Inference]],
+    y_metrics: List[Union[TestSampleMetric, Inference]],
+    binning_info: Optional[Tuple[float, float, float]] = None,  # start, end, num
+    x_logarithmic: bool = False,
+    y_logarithmic: bool = False,
+    metadata: bool = False,
+) -> CurvePlot:
```

Comment:
Unused?

Reply:
Metric vs metadata plot has been added in the latest commit